chrome/browser/android/adblock/adblock_bridge.cc

```diff
@@ -84,6 +84,35 @@ std::string replaceString(
     return subject;
 }
 
+std::string escapeJavaScriptString(const std::string& subject) {
+  std::string out;
+  for (const char& c : subject) {
+    switch (c) {
+      case '"':
+      case '\'':
+      case '\\':
+        out.append(1, '\\');
+        out.append(1, c);
+        break;
+
+      case '\n':
+        out.append("\\n");
+        break;
+
+      case '\r':
+        out.append("\\r");
+        break;
+
+      default:
+        out.append(1, c);
+    }
+  }
+
+  return replaceString(replaceString(out,
```

Comment:
As for me working with strings is more clear and readable instead of playing with bytes and index iterator like: `if length() > 3 and [i-2] == 0x.. and [i-1] == 0x..`.

Reply:
It will not work because UTF-8 string is a sequence of bytes, you need to search exactly for bytes, e.g. `0xE2 0x80 0xA8` then replace them by a sequence of ASCII characters `\u2028`.

Reply:
why isn't "u2028" not equal to `{ 0xE2, 0x80, 0xA8 }` byte array in internal representation ?

Reply:
`std::string` is just an array of bytes, the sub-string "u2028" is five bytes, 'u', '2' and so on, and these bytes are not the bytes we are looking for.